apis/ec2/v1beta1/zz_tag_types.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2022 Upbound Inc.
+*/
+
+// Code generated by upjet. DO NOT EDIT.
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	v1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+type TagObservation struct {
+
+	// EC2 resource identifier and key, separated by a comma (,)
+	ID *string `json:"id,omitempty" tf:"id,omitempty"`
+}
+
+type TagParameters struct {
+
+	// The tag name.
+	// +kubebuilder:validation:Required
+	Key *string `json:"key" tf:"key,omitempty"`
+
+	// Region is the region you'd like your resource to be created in.
+	// +upjet:crd:field:TFTag=-
+	// +kubebuilder:validation:Required
+	Region *string `json:"region" tf:"-"`
+
+	// The ID of the EC2 resource to manage the tag for.
+	// +crossplane:generate:reference:type=github.com/upbound/provider-aws/apis/ec2/v1beta1.VPNConnection
```

Comment:
Hi @haarchri & @ytsarev,
This reference is injected by the auto-reference injector in upjet because the provided example in Terraform registry makes use of a `aws_vpn_connection`:
https://registry.terraform.io/providers/hashicorp/aws/4.52.0/docs/resources/ec2_tag

In cases where the injected reference is not appropriate, we remove the injected reference with a configuration override. An example is here:
https://github.com/upbound/provider-aws/blob/e629ddeb0142fd1cc3aa1bd1207f3cf30db60222/config/cloudwatchevents/config.go#L38
, where `arn` is auto-injected and we remove it with a configuration override.

Reply:
thanks ;) was not aware of this option - updated the PR